www/lib/encryption/core.ts

```diff
@@ -0,0 +1,47 @@
+import { Signer } from 'ethers';
+import { signatureToPrivateKey } from 'rwtp';
+import nacl from 'tweetnacl';
+import { BoxKeyPair } from 'tweetnacl';
+
+// Creates an encryption key
+export async function createEncryptionKey(signer: Signer, scope: string) {
+  const sig = await signer.signMessage(scope);
+  return signatureToPrivateKey(sig);
+}
+
+export function keyAsHex(keypair: BoxKeyPair) {
+  return {
+    publicKeyAsHex: Buffer.from(keypair.publicKey).toString('hex'),
+    secretKeyAsHex: Buffer.from(keypair.secretKey).toString('hex'),
```

Comment:
I'd remove the secret key here to avoid potential misuse. The secret key should only be used for decryption, and it doesn't need to be formatted in hex for that purpose.